source/unified-test-format/unified-test-format.rst

```diff
@@ -959,6 +1050,95 @@ The structure of this object is as follows:
+
+- ``poolReadyEvent``: Optional object. If present, this object MUST be an
+  empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_poolClearedEvent:
+
+- ``poolClearedEvent``: Optional object. Assertions for one or more
+  `PoolClearedEvent <../connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#events>`__
+  fields.
+
+  The structure of this object is as follows:
+
+  - ``serverId``: Defined in `hasServerId`_.
+
+.. _expectedEvent_poolClosedEvent:
+
+- ``poolClosedEvent``: Optional object. If present, this object MUST be an
+  empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionCreatedEvent:
+
+- ``connectionCreatedEvent``: Optional object. If present, this object MUST be
+  an empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionReadyEvent:
+
+- ``connectionReadyEvent``: Optional object. If present, this object MUST be an
+  empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionClosedEvent:
+
+- ``connectionClosedEvent``: Optional object. Assertions for one or more
+  `ConnectionClosedEvent <../connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#events>`__
+  fields.
+
+  The structure of this object is as follows:
+
+  - ``reason``: Optional string. Test runners MUST assert that the reason in the
+    published event matches this value. Valid values for this field are defined
+    in the CMAP spec.
+
+.. _expectedEvent_connectionCheckOutStartedEvent:
+
+- ``connectionCheckOutStartedEvent``: Optional object. If present, this object
+  MUST be an empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionCheckOutFailedEvent:
+
+- ``connectionCheckOutFailedEvent``: Optional object. Assertions for one or more
+  `ConnectionCheckOutFailedEvent
+  <../connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#events>`__
+  fields.
+
+  The structure of this object is as follows:
+
+  - ``reason``: Optional string. Test runners MUST assert that the reason in the
+    published event matches this value. Valid values for this field are defined
+    in the CMAP spec.
+
+.. _expectedEvent_connectionCheckedOutEvent:
+
+- ``connectionCheckedOutEvent``: Optional object. If present, this object
+  MUST be an empty document as all fields in this event are non-deterministic.
+
+.. _expectedEvent_connectionCheckedInEvent:
+
+- ``connectionCheckedInEvent``: Optional object. If present, this object
+  MUST be an empty document as all fields in this event are non-deterministic.
+
+
+hasServerId
+```````````
+
+This field is an optional boolean that specifies whether or not the ``serverId``
+field of an event is set. If true, test runners MUST assert that the field is
+set and is a non-empty BSON ObjectId (i.e. all bytes of the ObjectId are not 0).
+If false, test runners MUST assert that the field is not set.
```

Comment:
Should you clarify that "is not set" allows an ObjectId with all zero bytes? Or rather say "is not set or is an empty BSON ObjectId"?

Reply:
Done.